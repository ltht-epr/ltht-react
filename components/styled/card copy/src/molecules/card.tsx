/** @jsx jsx */
import React from 'react'
import { css, jsx, SerializedStyles } from '@emotion/core'

import { CSS_RESET, TEXT_COLOURS, CARD_BACKGROUND_COLOUR, CARD_BACKGROUND_COLOUR_NO_DATA } from '@ltht-react/styles'
import CardEmpty from '../atoms/card-empty'

const styles = (noData: boolean): SerializedStyles => {
  return css`
  ${CSS_RESET}
  background: ${noData ? CARD_BACKGROUND_COLOUR_NO_DATA : CARD_BACKGROUND_COLOUR};
  color: ${noData ? TEXT_COLOURS.SECONDARY : TEXT_COLOURS.PRIMARY};
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 
    0px 2px 1px -1px rgba(102, 102, 102, 0.1), 
    0px 1px 1px 0px rgba(102, 102, 102, 0.15), 
    0px 1px 3px 0px rgba(102, 102, 102,.6);
  -webkit-font-smoothing: antialiased;
`
}

const Card: React.FC<Props> = ({ children, noData = false }) => {
  return (
    <div css={styles(noData)}>
      {children}
      {noData && <CardEmpty />}
    </div>
  )
}

interface Props {
  noData?: boolean
}

export default Card
