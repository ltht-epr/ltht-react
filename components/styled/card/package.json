{
  "name": "@ltht-react/card",
  "version": "2.0.3",
  "description": "ltht-react styled Card component.",
  "author": "LTHT",
  "homepage": "",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "files": [
    "lib",
    "src"
  ],
  "directories": {
    "lib": "lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn build:lib",
    "watch": "tsc -w",
    "build:lib": "tsc",
    "type:check": "tsc --noEmit",
    "lint": "prettylint 'src/*.{ts,tsx}'"
  },
  "dependencies": {
    "@emotion/react": "^11.0.0",
    "@emotion/styled": "^11.0.0",
    "@ltht-react/banner": "^2.0.3",
    "@ltht-react/icon": "^2.0.3",
    "@ltht-react/styles": "^2.0.3",
    "@ltht-react/types": "^2.0.3",
    "@ltht-react/utils": "^2.0.3",
    "classnames": "^2.2.6",
    "react": "^18.2.0"
  },
  "gitHead": "4d364556bb11e294d86697c2dac66742f7aabc7f"
}
