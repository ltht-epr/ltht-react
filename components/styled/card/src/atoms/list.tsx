import React from 'react'
import classNames from 'classnames'
import styled from '@emotion/styled'

export interface Props {
  classes?: string
}

const StyledList = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-width: 1px 0 1px 0;

  &:first-child {
    border-top-width: 0;
    border-top-left-radius: inherit;
    border-top-right-radius: inherit;
  }

  &:last-child {
    border-bottom-width: 0;
    border-bottom-left-radius: inherit;
    border-bottom-right-radius: inherit;
  }
`

const List: React.FC<Props> = ({ classes, children }) => {
  return <StyledList className={classNames('card__list', classes)}>{children}</StyledList>
}

export default List
