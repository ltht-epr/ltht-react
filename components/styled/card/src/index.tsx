export * from './atoms'
export { default as Card } from './molecules/card'
