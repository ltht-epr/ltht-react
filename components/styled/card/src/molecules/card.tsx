import React from 'react'
import classNames from 'classnames'
import styled from '@emotion/styled'
import { CSS_RESET, TEXT_COLOURS, CARD_BACKGROUND_COLOUR } from '@ltht-react/styles'
import Alert, { Props as AlertProps } from '../atoms/alert'
import Body, { Props as BodyProps } from '../atoms/body'
import Footer, { Props as FooterProps } from '../atoms/footer'
import Header, { Props as HeaderProps } from '../atoms/header'
import ListItem, { Props as ListItemProps } from '../atoms/list-item'
import List, { Props as ListProps } from '../atoms/list'
import Subtitle, { Props as SubtitleProps } from '../atoms/subtitle'
import Text, { Props as TextProps } from '../atoms/text'
import Title, { Props as TitleProps } from '../atoms/title'

interface CardComposition {
  Alert: React.FC<AlertProps>
  Body: React.FC<BodyProps>
  Footer: React.FC<FooterProps>
  Header: React.FC<HeaderProps>
  ListItem: React.FC<ListItemProps>
  List: React.FC<ListProps>
  Subtitle: React.FC<SubtitleProps>
  Text: React.FC<TextProps>
  Title: React.FC<TitleProps>
}

type TextAlignValues = 'left' | 'center' | 'right'

interface StyledProps {
  textAlign: TextAlignValues
}

interface Props {
  textAlign?: TextAlignValues
  classes?: string
}

const StyledCard = styled.div`
  ${CSS_RESET}
  position: relative;
  display: flex;
  flex-direction: column;
  text-align: ${(props: StyledProps): string => props.textAlign};
  background: ${CARD_BACKGROUND_COLOUR};
  color: ${TEXT_COLOURS.PRIMARY};
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102, 102, 102, 0.1), 0px 1px 1px 0px rgba(102, 102, 102, 0.15),
    0px 1px 3px 0px rgba(102, 102, 102, 0.6);
  -webkit-font-smoothing: antialiased;

  & > .card__alert + .card__list,
  > .card__alert + .card__body,
  > .card__list + .card__footer {
    border-top-width: 0;
  }
`

const Card: React.FC<Props> & CardComposition = ({ textAlign = 'left', classes, children }) => {
  return (
    <StyledCard textAlign={textAlign} className={classNames('card', classes)}>
      {children}
    </StyledCard>
  )
}

Card.Alert = Alert
Card.Body = Body
Card.Footer = Footer
Card.Header = Header
Card.ListItem = ListItem
Card.List = List
Card.Subtitle = Subtitle
Card.Text = Text
Card.Title = Title

export default Card
