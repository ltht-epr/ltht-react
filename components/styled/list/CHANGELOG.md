# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 1.3.66 (2022-04-26)

**Note:** Version bump only for package @ltht-react/list





## 1.3.65 (2022-04-25)

**Note:** Version bump only for package @ltht-react/list





## 1.3.64 (2022-04-22)

**Note:** Version bump only for package @ltht-react/list





## 1.3.63 (2022-03-31)

**Note:** Version bump only for package @ltht-react/list





## 1.3.62 (2022-03-22)

**Note:** Version bump only for package @ltht-react/list





## 1.3.61 (2022-03-21)

**Note:** Version bump only for package @ltht-react/list





## 1.3.60 (2022-03-18)

**Note:** Version bump only for package @ltht-react/list





## 1.3.59 (2022-03-18)

**Note:** Version bump only for package @ltht-react/list





## 1.3.58 (2022-03-10)

**Note:** Version bump only for package @ltht-react/list





## 1.3.57 (2022-03-08)

**Note:** Version bump only for package @ltht-react/list





## 1.3.56 (2022-03-02)

**Note:** Version bump only for package @ltht-react/list





## 1.3.55 (2022-02-17)

**Note:** Version bump only for package @ltht-react/list





## 1.3.54 (2022-02-16)

**Note:** Version bump only for package @ltht-react/list





## 1.3.53 (2022-02-15)

**Note:** Version bump only for package @ltht-react/list





## 1.3.52 (2022-02-11)

**Note:** Version bump only for package @ltht-react/list





## 1.3.51 (2022-02-11)

**Note:** Version bump only for package @ltht-react/list





## 1.3.50 (2022-02-10)

**Note:** Version bump only for package @ltht-react/list





## 1.3.49 (2022-02-09)

**Note:** Version bump only for package @ltht-react/list





## 1.3.48 (2022-02-07)

**Note:** Version bump only for package @ltht-react/list





## 1.3.47 (2022-02-02)

**Note:** Version bump only for package @ltht-react/list





## 1.3.46 (2022-02-02)

**Note:** Version bump only for package @ltht-react/list





## 1.3.45 (2022-01-25)

**Note:** Version bump only for package @ltht-react/list





## 1.3.44 (2022-01-21)

**Note:** Version bump only for package @ltht-react/list





## 1.3.43 (2022-01-19)

**Note:** Version bump only for package @ltht-react/list





## 1.3.42 (2022-01-19)

**Note:** Version bump only for package @ltht-react/list





## 1.3.41 (2022-01-18)

**Note:** Version bump only for package @ltht-react/list





## 1.3.40 (2022-01-17)

**Note:** Version bump only for package @ltht-react/list





## 1.3.39 (2022-01-11)

**Note:** Version bump only for package @ltht-react/list





## 1.3.38 (2022-01-10)

**Note:** Version bump only for package @ltht-react/list





## 1.3.37 (2022-01-07)

**Note:** Version bump only for package @ltht-react/list





## 1.3.36 (2022-01-07)

**Note:** Version bump only for package @ltht-react/list





## 1.3.35 (2022-01-06)

**Note:** Version bump only for package @ltht-react/list





## 1.3.34 (2022-01-05)

**Note:** Version bump only for package @ltht-react/list





## 1.3.33 (2022-01-05)

**Note:** Version bump only for package @ltht-react/list





## 1.3.32 (2022-01-04)

**Note:** Version bump only for package @ltht-react/list





## 1.3.31 (2021-12-16)

**Note:** Version bump only for package @ltht-react/list





## 1.3.30 (2021-12-02)

**Note:** Version bump only for package @ltht-react/list





## 1.3.29 (2021-12-02)

**Note:** Version bump only for package @ltht-react/list





## 1.3.28 (2021-12-01)

**Note:** Version bump only for package @ltht-react/list





## 1.3.27 (2021-11-24)

**Note:** Version bump only for package @ltht-react/list





## 1.3.26 (2021-11-24)

**Note:** Version bump only for package @ltht-react/list





## 1.3.25 (2021-11-24)

**Note:** Version bump only for package @ltht-react/list





## 1.3.24 (2021-11-23)

**Note:** Version bump only for package @ltht-react/list





## 1.3.23 (2021-10-22)

**Note:** Version bump only for package @ltht-react/list





## 1.3.22 (2021-10-19)

**Note:** Version bump only for package @ltht-react/list





## 1.3.21 (2021-10-13)

**Note:** Version bump only for package @ltht-react/list





## 1.3.20 (2021-10-05)

**Note:** Version bump only for package @ltht-react/list





## 1.3.19 (2021-10-05)

**Note:** Version bump only for package @ltht-react/list





## 1.3.18 (2021-09-30)

**Note:** Version bump only for package @ltht-react/list





## 1.3.17 (2021-09-29)

**Note:** Version bump only for package @ltht-react/list





## 1.3.16 (2021-09-28)

**Note:** Version bump only for package @ltht-react/list





## 1.3.15 (2021-09-22)

**Note:** Version bump only for package @ltht-react/list





## 1.3.14 (2021-09-21)

**Note:** Version bump only for package @ltht-react/list





## 1.3.13 (2021-09-20)

**Note:** Version bump only for package @ltht-react/list





## 1.3.12 (2021-09-17)

**Note:** Version bump only for package @ltht-react/list





## 1.3.11 (2021-09-17)

**Note:** Version bump only for package @ltht-react/list





## 1.3.10 (2021-08-26)

**Note:** Version bump only for package @ltht-react/list





## 1.3.9 (2021-08-23)

**Note:** Version bump only for package @ltht-react/list





## [1.3.8](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.7...@ltht-react/list@1.3.8) (2021-08-02)

**Note:** Version bump only for package @ltht-react/list





## [1.3.7](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.6...@ltht-react/list@1.3.7) (2021-07-14)

**Note:** Version bump only for package @ltht-react/list





## [1.3.6](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.5...@ltht-react/list@1.3.6) (2021-07-12)

**Note:** Version bump only for package @ltht-react/list





## [1.3.5](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.4...@ltht-react/list@1.3.5) (2021-07-06)

**Note:** Version bump only for package @ltht-react/list





## [1.3.4](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.3...@ltht-react/list@1.3.4) (2021-06-29)

**Note:** Version bump only for package @ltht-react/list





## [1.3.3](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.2...@ltht-react/list@1.3.3) (2021-06-21)

**Note:** Version bump only for package @ltht-react/list





## [1.3.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@1.3.0...@ltht-react/list@1.3.1) (2021-04-23)

**Note:** Version bump only for package @ltht-react/list





## [0.1.26](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.24...@ltht-react/list@0.1.26) (2020-11-19)

**Note:** Version bump only for package @ltht-react/list

## [0.1.25](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.24...@ltht-react/list@0.1.25) (2020-10-29)

**Note:** Version bump only for package @ltht-react/list

## [0.1.24](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.23...@ltht-react/list@0.1.24) (2020-09-10)

**Note:** Version bump only for package @ltht-react/list

## [0.1.23](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.22...@ltht-react/list@0.1.23) (2020-09-01)

**Note:** Version bump only for package @ltht-react/list

## [0.1.22](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.21...@ltht-react/list@0.1.22) (2020-08-21)

**Note:** Version bump only for package @ltht-react/list

## [0.1.21](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.20...@ltht-react/list@0.1.21) (2020-08-19)

**Note:** Version bump only for package @ltht-react/list

## [0.1.20](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.19...@ltht-react/list@0.1.20) (2020-08-19)

**Note:** Version bump only for package @ltht-react/list

## [0.1.19](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.18...@ltht-react/list@0.1.19) (2020-08-18)

**Note:** Version bump only for package @ltht-react/list

## [0.1.18](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.17...@ltht-react/list@0.1.18) (2020-08-11)

**Note:** Version bump only for package @ltht-react/list

## [0.1.17](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.16...@ltht-react/list@0.1.17) (2020-08-10)

**Note:** Version bump only for package @ltht-react/list

## [0.1.16](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.15...@ltht-react/list@0.1.16) (2020-08-05)

**Note:** Version bump only for package @ltht-react/list

## [0.1.15](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.14...@ltht-react/list@0.1.15) (2020-08-05)

**Note:** Version bump only for package @ltht-react/list

## [0.1.14](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.13...@ltht-react/list@0.1.14) (2020-08-05)

**Note:** Version bump only for package @ltht-react/list

## [0.1.12](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.11...@ltht-react/list@0.1.12) (2020-08-04)

**Note:** Version bump only for package @ltht-react/list

## [0.1.11](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.10...@ltht-react/list@0.1.11) (2020-05-06)

**Note:** Version bump only for package @ltht-react/list

## [0.1.10](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.9...@ltht-react/list@0.1.10) (2020-04-14)

**Note:** Version bump only for package @ltht-react/list

## [0.1.9](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.8...@ltht-react/list@0.1.9) (2020-04-09)

**Note:** Version bump only for package @ltht-react/list

## [0.1.8](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.7...@ltht-react/list@0.1.8) (2020-04-08)

**Note:** Version bump only for package @ltht-react/list

## [0.1.7](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.6...@ltht-react/list@0.1.7) (2020-04-03)

**Note:** Version bump only for package @ltht-react/list

## [0.1.6](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.5...@ltht-react/list@0.1.6) (2020-03-16)

**Note:** Version bump only for package @ltht-react/list

## [0.1.5](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.4...@ltht-react/list@0.1.5) (2020-03-12)

**Note:** Version bump only for package @ltht-react/list

## [0.1.4](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.3...@ltht-react/list@0.1.4) (2020-03-06)

**Note:** Version bump only for package @ltht-react/list

## [0.1.3](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.2...@ltht-react/list@0.1.3) (2020-03-02)

**Note:** Version bump only for package @ltht-react/list

## [0.1.2](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.1...@ltht-react/list@0.1.2) (2020-02-27)

**Note:** Version bump only for package @ltht-react/list

## [0.1.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/list@0.1.0...@ltht-react/list@0.1.1) (2020-02-26)

**Note:** Version bump only for package @ltht-react/list
