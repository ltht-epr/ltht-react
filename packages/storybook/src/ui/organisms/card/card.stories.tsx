import { Story } from '@storybook/react'

import Card from '@ltht-react/card'
import AllergySummary from '@ltht-react/allergy-summary'
import { InfoSummary, WarningSummary, ErrorSummary, MissingDataSummary } from '@ltht-react/type-summary'
import allergies from '../../../clinical/organisms/allergies/allergies.fixtures'

const clickHandler = (): void => {
  // eslint-disable-next-line no-console
  console.log('Handler clicked')
}

export const Basic: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.Body>
      <Card.Subtitle>Subtitle</Card.Subtitle>
      <Card.Text>Body Text</Card.Text>
    </Card.Body>
  </Card>
)

export const WithFooter: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.Body>
      <Card.Subtitle>Subtitle</Card.Subtitle>
      <Card.Text>Body Text</Card.Text>
    </Card.Body>
    <Card.Footer>Footer</Card.Footer>
  </Card>
)

export const WithList: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.List>
      <Card.ListItem>Item 1</Card.ListItem>
      <Card.ListItem>Item 2</Card.ListItem>
    </Card.List>
  </Card>
)

export const ClickableList: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.List>
      <Card.ListItem onClick={clickHandler}>
        <div>Item 1</div>
      </Card.ListItem>
      <Card.ListItem onClick={clickHandler}>
        <div>Item 2</div>
      </Card.ListItem>
    </Card.List>
  </Card>
)

export const InfoSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <InfoSummary clickHandler={clickHandler} />
    <Card.List>
      {allergies.map((allergy) => (
        <Card.ListItem key={allergy.id}>
          <AllergySummary allergy={allergy} />
        </Card.ListItem>
      ))}
    </Card.List>
    <InfoSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>Missing 10</Card.ListItem>
    </Card.List>
    <WarningSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>Missing 10</Card.ListItem>
      <Card.ListItem>Missing 10</Card.ListItem>
    </Card.List>
  </Card>
)
InfoSummaryStory.storyName = 'Info Summary'

export const WarningSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <WarningSummary clickHandler={clickHandler} />
    <Card.List>
      {allergies.map((allergy) => (
        <Card.ListItem key={allergy.id}>
          <AllergySummary allergy={allergy} />
        </Card.ListItem>
      ))}
    </Card.List>
  </Card>
)
WarningSummaryStory.storyName = 'Warning Summary'

export const ErrorSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <ErrorSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>No Data</Card.ListItem>
    </Card.List>
  </Card>
)
ErrorSummaryStory.storyName = 'Error Summary'

export const MissingDataSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <MissingDataSummary
      type="allergies"
      noData={['Item 1', 'Item 2']}
      missingData={['Item 1', 'Item 2']}
      partialData={['Item 1', 'Item 2']}
      completeData={['Item 1', 'Item 2']}
    />
  </Card>
)
MissingDataSummaryStory.storyName = 'Missing Data Summary'

export const NoData: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
  </Card>
)

export default { title: 'UI/Organisms/Card' }
