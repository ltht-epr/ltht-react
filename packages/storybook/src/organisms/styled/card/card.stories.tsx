import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import readme from '@ltht-react/card/README.md'
import { ListItem } from '@ltht-react/list'
import { MissingDataSummary, MissingDataWarning } from '@ltht-react/summary'
import { Card, CardHeader, CardBody, CardFooter, CardList } from '@ltht-react/card'

const stories = storiesOf('Organisms - Styled|Card', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Basic', () => (
  <Card>
    <CardHeader>
      <h3>Header</h3>
    </CardHeader>
    <CardBody>Body</CardBody>
  </Card>
))

stories.addWithJSX('Footer', () => (
  <Card>
    <CardHeader>
      <h3>Header</h3>
    </CardHeader>
    <CardBody>Body</CardBody>
    <CardFooter>Footer</CardFooter>
  </Card>
))

stories.addWithJSX('List', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList>
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('List (Clickable)', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('Missing Data Warning', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <MissingDataWarning />
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('Missing Data Summary', () => (
  <div>
    <Card>
      <CardHeader position="center">
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <MissingDataSummary
            noData={['Item 1', 'Item 2']}
            partialData={['Item 3', 'Item 4']}
            allData={['Item 5', 'Item 6']}
          />
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('No Data', () => (
  <div>
    <Card noData>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
    </Card>
  </div>
))
