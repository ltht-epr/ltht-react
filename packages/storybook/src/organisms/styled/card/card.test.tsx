import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import { ListItem } from '@ltht-react/list'
import { Card, CardBody, CardHeader, CardFooter, CardList } from '@ltht-react/card'

const components = (
  <>
    <Card>
      <CardHeader>
        <h3>Heading</h3>
      </CardHeader>
      <CardBody>Body</CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
    <Card>
      <CardHeader>
        <h3>Heading</h3>
      </CardHeader>
      <CardList>
        <ListItem>Item 1</ListItem>
        <ListItem>Item 2</ListItem>
      </CardList>
      <CardFooter>Footer</CardFooter>
    </Card>
  </>
)

describe('Card', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(components, div)
  })
  it('matches snapshot', () => {
    expect(mount(components)).toMatchSnapshot('wrapper mount')
  })
})
