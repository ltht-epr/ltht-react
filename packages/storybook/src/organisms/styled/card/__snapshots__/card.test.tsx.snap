// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot: wrapper mount 1`] = `
Array [
  .emotion-3 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #FFFFFF;
  color: #231f20;
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102,102,102,0.1), 0px 1px 1px 0px rgba(102,102,102,0.15), 0px 1px 3px 0px rgba(102,102,102,.6);
  -webkit-font-smoothing: antialiased;
}

.emotion-3 *,
.emotion-3 ::before,
.emotion-3 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-3 * {
  margin: 0;
  padding: 0;
}

.emotion-3 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    font-size: 0.9rem;
  }

  .emotion-3 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-3 {
    font-size: 1rem;
  }

  .emotion-3 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-3 {
    font-size: 1.1rem;
  }

  .emotion-3 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem !important;
}

.emotion-1 {
  padding: 0 0.5rem 0 0.5rem !important;
}

.emotion-1 > dl:first-of-type {
  margin-top: 0 !important;
}

.emotion-2 {
  padding: 0 0.5rem 0 0.5rem !important;
}

<Card>
    <div
      className="emotion-3"
    >
      <CardHeader>
        <div
          className="emotion-0"
          onClick={[Function]}
          title=""
        >
          <h3>
            Heading
          </h3>
        </div>
      </CardHeader>
      <CardBody>
        <div
          className="emotion-1"
        >
          Body
        </div>
      </CardBody>
      <CardFooter>
        <div
          className="emotion-2"
        >
          Footer
        </div>
      </CardFooter>
    </div>
  </Card>,
  .emotion-3 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #FFFFFF;
  color: #231f20;
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102,102,102,0.1), 0px 1px 1px 0px rgba(102,102,102,0.15), 0px 1px 3px 0px rgba(102,102,102,.6);
  -webkit-font-smoothing: antialiased;
}

.emotion-3 *,
.emotion-3 ::before,
.emotion-3 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-3 * {
  margin: 0;
  padding: 0;
}

.emotion-3 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    font-size: 0.9rem;
  }

  .emotion-3 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-3 {
    font-size: 1rem;
  }

  .emotion-3 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-3 {
    font-size: 1.1rem;
  }

  .emotion-3 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem !important;
}

.emotion-2 {
  padding: 0 0.5rem 0 0.5rem !important;
}

.emotion-1 {
  list-style: none;
}

.emotion-1 li {
  border-top: 1px solid #d8d8d8;
  padding: 0.5rem 0.5rem 0.5rem 0;
  margin-right: -0.5rem;
}

.emotion-1 #banner {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.emotion-1 #banner + li {
  border-top: 0px !important;
}

<Card>
    <div
      className="emotion-3"
    >
      <CardHeader>
        <div
          className="emotion-0"
          onClick={[Function]}
          title=""
        >
          <h3>
            Heading
          </h3>
        </div>
      </CardHeader>
      <CardList>
        <ul
          className="emotion-1"
        >
          <ListItem>
            <li>
              Item 1
            </li>
          </ListItem>
          <ListItem>
            <li>
              Item 2
            </li>
          </ListItem>
        </ul>
      </CardList>
      <CardFooter>
        <div
          className="emotion-2"
        >
          Footer
        </div>
      </CardFooter>
    </div>
  </Card>,
]
`;
